src/Foundation/Console/KeyGenerateCommand.php

```diff
@@ -85,12 +112,36 @@ protected function setKeyInConfigFile($key)
      *
      * @return array
      */
-    protected function getKeyFile()
+    protected function getKeyFile(): array
     {
-        $env = $this->option('env') ? $this->option('env').'/' : '';
+        $env = $this->option('env') ? $this->option('env').'/' : $this->laravel->environment().'/';
+
+        $path = $this->laravel['path.config']."/{$env}app.php";
+
+        if ($this->files->exists($path)) {
+            $config = include $path;
+
+            if (!isset($config['key'])) {
+                $env = '';
+            }
+        } else {
+            $env = '';
+        }
 
         $contents = $this->files->get($path = $this->laravel['path.config']."/{$env}app.php");
 
         return [$path, $contents];
     }
+
+    /**
+     * Get a regex pattern that will match config APP_KEY with any random key
+     *
+     * @return string
+     */
+    protected function configKeyReplacementPattern(): string
```

Comment:
Isn't there a ConfigWriter class built into October? Why are we reinventing the wheel here?

Reply:
I was not aware of the config writer class, but I tried using it just now and it looks like it will need to be modified in order to support updating values which contain an env call `'key' => env('APP_KEY', 'CHANGE_ME!'),` since it only works with regular key => value lines. `'key' => 'CHANGE_ME',`

I'll try and see if there's anything from this PR that can be used: https://github.com/daftspunk/laravel-config-writer/pull/22/files or maybe this can just be merged as-is if there aren't any changes needed.